Array_Vector_Exercises/Exercise1/Menu.cpp

```diff
@@ -0,0 +1,380 @@
+			std::cout << "1 - Circle" << std::endl;
+			std::cout << "2 - Square" << std::endl;
+			std::cout << "-----------------------------------------------" << std::endl;
+			std::cout << "Enter a valid option: ";
+
+			int KindShape;
+			KindShape = ValidateInput(UserInput, 1, 2);
+
+			if (KindShape == 1)
+			{
+				std::system("cls");
+				std::cout << "Insert circle's Radius: ";
+
+				float InRadius;
+				InRadius = ValidateInputFloat(InRadius);
+				Shapes.PushBack(new FCircle(InRadius));
+			}
+
+			else
+			{
+				std::system("cls");
+				std::cout << "Insert square's lenght: ";
+				float InLenght;
+				InLenght = ValidateInputFloat(InLenght);
+
+				std::cout << "Insert square's heignt: ";
+				float InHeight;
+				InHeight = ValidateInputFloat(InHeight);
+
+				Shapes.PushBack(new FSquare(InLenght, InHeight));
+			}
+			std::cout << "You shape was created succefully" << std::endl;
+		}
+
+		else if (UserInput == 2)
+		{
+
+			if (Shapes.IsEmpty())
+			{
+				std::system("cls");
+				std::cout << "You don't create a shape yet..." << std::endl;
+			}
+
+			else
+			{
+				std::system("cls");
+				std::cout << "These are the shapes that you created: " << std::endl;
+
+				for (int i = 0; i < Shapes.GetSize(); i++)
+				{
+					std::cout << "--------------------------------- " << std::endl;
+					std::cout << "Shape #" << i + 1 << std::endl;
+					PrintArray(*Shapes[i]);
+				}
+			}
+
+			std::cout << "Press 1... to back previus menu" << std::endl;
+			int MenuInput;
+			std::cin >> MenuInput;
+
+			while (std::cin.fail() || MenuInput != 1)
+			{
+				std::cin.clear();
+				std::cin.ignore(100, '\n');
+				std::cout << "Invalid Option, please enter a valid Option " << std::endl;
+				std::cin >> MenuInput;
+				if (MenuInput == 1)
+				{
+					ActiveMenuCommand = EMenuCommand::DynamicArray;
+				}
+			}
+		}
+
+		else
+		{
+			//freeing Memory
+			for (int i = 0; i < Shapes.GetSize(); i++)
+			{
+				delete Shapes[i];
+			}
+			ActiveMenuCommand = EMenuCommand::MainMenu;
+		}
+
+	} while (UserInput != 3);
+}
+
+FShape* FMenu::CreateShape()
+{
+	std::cout << "-------------------------------------" << std::endl;
+	std::cout << "1- Circle" << std::endl;
+	std::cout << "2- Square" << std::endl;
+	std::cout << "-------------------------------------" << std::endl;
+
+	int UserInput;
+	UserInput = ValidateInput(UserInput, 1, 2);
+
+	if (UserInput == 1)
+	{
+		std::cout << "Insert circle's Radius: ";
+		float InRadius;
+		
+		InRadius = ValidateInputFloat(InRadius);
+		return new FCircle(InRadius);
+		
+			
+	}
+	else
+	{
+		std::cout << "Insert square's lenght: ";
+		float InLenght;
+		InLenght = ValidateInputFloat(InLenght);
+
+		std::cout << "Insert square's heignt: ";
+		float InHeight;
+		InHeight = ValidateInputFloat(InHeight);
+
+		return new FSquare(InLenght, InHeight);
+	}
+}
+
+//---------------------------------------------------------------------------------
+void FMenu::PrintArray(const FShape& Shape) const
```

Comment:
Why is this called print array if it prints a single shape?

Reply:
Also, this works correctly, but since we're manipulating FShaps with pointers, it may be simpler to use a shape pointer as a parameter. What do you think?